src/Parser/TemplateParser.cs

```diff
@@ -138,22 +139,90 @@ private string ParseName(out int parameterIndex)
                 c = Peek();
                 if (c == '}' || c == ':' || c == ',')
                     parameterIndex = parsed;
-                _position = start;        
+                _position = start;
             }
 
             if (parameterIndex == -1)
                 _isPositional = false;
 
-            return ReadUntil(HoleDelimiters);            
+            return ReadUntil(HoleDelimiters);
         }
 
+        /// <summary>
+        /// Parse format after hole name/index. Handle the escaped { and } in the format. Don't read the last }
+        /// </summary>
+        /// <returns></returns>
         private string ParseFormat()
         {
+
             Skip(':');
-            // TODO: Escaped }} in formats?
-            return ReadUntil('}');
+            string format = null;
+            while (true)
+            {
+                if (format == null)
```

Comment:
You don't really need the condition if you put 
`string format = ReadUntil(TextDelimiters)`
outside the loop and then move
`format += ReadUntil(TextDelimiters)` 
at the end of the loop.